src/game.js

```diff
@@ -209,6 +218,14 @@
             // reset the frame counter
             frameCounter = 0;
             frameRate = ~~(0.5 + 60 / me.sys.fps);
+            
+            // set step size based on the updatesPerSecond
+            stepSize = (1000 / Math.min(me.sys.updatesPerSecond, me.sys.fps));
```

Comment:
This comment: https://github.com/melonjs/melonJS/issues/99#issuecomment-126942149 For collision accuracy (e.g resolving issue #16) the collision detection is done at a higher frequency than drawing, giving more intermediate states, preventing high-velocity entities from skipping collisions that "obviously" should occur.

That's why you would want to run `update` multiple times per frame.

However, you still haven't addressed my comment that the while-statement never loops? It only executes zero or one time, which is the behavior of an if-statement.